lib/storage/mongo.js

```diff
@@ -250,6 +258,74 @@ class Mongo {
             console.error('saveMonitor error: ' + error.message)
         }
     }
+
+
+    async getApplication(data = {}) {
+        try {
+            const {name, version, path} = data
+            console.log(name, version)
+            if(!name){
+                throw new Error("The name of the application must be valid")
+            }
+            let query = (name) ? { name } : {}
+            if (version)
+                query = { name, version }
+            console.log("Query:", query)
+            const doc = await this._applicationStorage.find(query)
+            return doc;
+        } catch (error) {
+            console.error('getApplication error: ' + error.message)
+            throw error.message;
+        }
+    }
+    
+    async getLatestVersion(data = {})
+    {
+        try {
+            const {name} = data
+            if(!name){
+                throw new Error("The name of the application must be valid")
+            }
+            const query = { name }
+            const doc = await this._applicationStorage.find(query).sort({ version: 'desc' }).limit(1);
+            return doc
+        }
+        catch (error){
+            console.log("getLatestVersion Error - " + error.message)
+        }
+    }
+    async updateApplication(doc, data) {
+        const {name, path, version } = data
+        if(path)
+            doc.path = path
+            doc.markModified("path");
```

Comment:
from what I understand (about mongodb) markModified is only needed for types that are not known by mongo. Not needed for a simple String field